ws-nextjs-app/pages/[service]/av-embeds/parseAvRoute/index.ts

```diff
@@ -0,0 +1,179 @@
+
+const MEDIA_ID_REGEX = /^p[0-9a-z]{7,}/;
+
+// Language codes
+const LANGS = [
+  'am',
+  'ar',
+  'az',
+  'bn',
+  'cy',
+  'en',
+  'en-gb',
+  'es',
+  'fa',
+  'fr',
+  'gd',
+  'gu',
+  'ha',
+  'hi',
+  'id',
+  'ig',
+  'ja',
+  'ko',
+  'ky',
+  'mr',
+  'my',
+  'ne',
+  'om',
+  'pa',
+  'pcm',
+  'ps',
+  'pt',
+  'pt-br',
+  'ru',
+  'ru-uk',
+  'ru-ua',
+  'rw',
+  'si',
+  'so',
+  'sr-cyrl',
+  'sr-latn',
+  'sw',
+  'ta',
+  'te',
+  'tg',
+  'th',
+  'ti',
+  'tr',
+  'uk',
+  'ur',
+  'uz',
+  'uz-cyrl',
+  'vi',
+  'yo',
+  'zh-hans',
+  'zh-hant',
+];
+
+const LANGS_REGEX = new RegExp(`^(${LANGS.join('|')})$`);
+
+const SERVICES = Object.keys(services) as Services[];
+const VARIANTS = ['lat', 'cyr', 'trad', 'simp'] as Variants[];
+
+const extractService = (query: Query): Services | null => {
+  const service = SERVICES.find(s => s !== 'ws' && query?.includes(s));
+
+  return service ?? null;
+};
+
+const extractVariant = (query: Query): Variants | null => {
+  const variant = VARIANTS.find(v => query?.includes(v));
+
+  return variant ?? null;
+};
+
+const extractPlatform = (query: Query): Platform | null => {
+  let platform: Platform | null = null;
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const id of query ?? []) {
+    if (CPS_ID_REGEX.test(id)) {
+      platform = 'cps';
+      break;
+    }
+    if (OPTIMO_ID_REGEX.test(id)) {
+      platform = 'optimo';
+      break;
+    }
+    if (TIPO_ID_REGEX.test(id)) {
+      platform = 'tipo';
+      break;
+    }
+  }
+
+  return platform;
+};
+
+const extractAssetId = (query: Query) => {
+  const assetId = query?.find((id: string) => {
+    return (
+      CPS_ID_REGEX.test(id) ||
+      OPTIMO_ID_REGEX.test(id) ||
+      TIPO_ID_REGEX.test(id)
+    );
+  });
+
+  return assetId ?? null;
+};
+
+const extractMediaId = (query: Query) => {
+  const mediaId = query?.find((id: string) => MEDIA_ID_REGEX.test(id));
+
+  return mediaId ?? null;
+};
+
+const extractLang = (query: Query) => {
+  const lang = query?.find((l: string) => LANGS_REGEX.test(l));
+
+  return lang ?? null;
+};
+
+const extractAmp = (query: Query) => {
+  const amp = query?.includes('amp');
+
+  return amp ?? null;
+};
+
+/**
+ *  Syndication route patterns:
+ *  -/:service/av-embeds/:asset_id
+ *  -/:service/av-embeds/:asset_uri_wo_service
+ *  -/:service/av-embeds/:asset_id/vpid/:vpid
+ *  -/:service/av-embeds/:asset_uri_wo_service/pid/:pid
+ *
+ *  Syndication route examples:
+ *  -/news/av-embeds/67303123
+ *  -/serbian/cyr/av-embeds/srbija-68707945
+ *  -/russian/av-embeds/38886884/vpid/p04s97g7
+ *  -/news/av-embeds/58228280/pid/p09s9t1j
+ */
+export default function parseAvRoute(resolvedUrl: string) {
+  const resolvedUrlWithoutQuery = resolvedUrl.split('?')?.[0];
+
+  const query = resolvedUrlWithoutQuery.split(/[/.]/).filter(Boolean);
+
+  // Assumes /ws/ routes are purely for Simorgh AMP pages
+  // - only for testing
+  const isSyndicationRoute = !query.includes('ws');
```

Comment:
I know this is still a draft / WIP, but wondering what is happening with AMP + Media Player? Or do we think that there might still be a need for an AMP version of the av-embed?

Reply:
Yea this is focused mainly on syndication routes for now. The `ws/av-embeds` routes will need to be supported for AMP media players in Article pages. The logic here does support those routes too (at least parsing out the values from the URL), so they are pretty much good to go.